src/java.base/share/classes/java/lang/System.java

```diff
@@ -2660,6 +2662,20 @@ public StackWalker newStackWalkerInstance(Set<StackWalker.Option> options,
                                                       Continuation continuation) {
                 return StackWalker.newInstance(options, null, contScope, continuation);
             }
+
+            @Override
+            public int mismatchUTF8(String str, byte[] b, int fromIndex, int toIndex) {
+                byte[] encoded = str.isLatin1() ? str.value() : str.getBytes(UTF_8.INSTANCE);
+                if (false) {
```

Comment:
Leaving the ArraySupport.mismatch code here for now if anyone wants to investigate the ~3% regression introduced by the range checks in Arrays.mismatch